rt/rs/security/http-signature/src/test/java/org/apache/cxf/rs/security/httpsignature/MessageVerifierTest.java

```diff
@@ -51,6 +51,7 @@
     private static final String METHOD = "GET";
     private static final String URI = "/test/signature";
     private static final String KEY_PAIR_GENERATOR_ALGORITHM = "RSA";
+    private static final String MESSAGE_BODY = "Hello";
```

Comment:
@AnilKumarHurkadli  I just realized there are no testcases for the case when message body is not present, exactly the flow you are implementing. Could you please add a couple please?

Reply:
@reta Sure will add the test case for which InvalidDataToVerifySignatureException will not be thrown if the body is null and passes fine. The issue was encountered specifically while testing DELETE method which "usually" do not have body and "Digest" header gets added by default in the required headers and is NOT removed as it was not satisfying below condition

if ((requestor && (status < 200 || status >= 300 || status == 204))
            || (!requestor && ("GET".equalsIgnoreCase(method) || "HEAD".equalsIgnoreCase(method)))) {
            signedHeaders.remove("digest");
        }

And was throwing InvalidDataToVerifySignatureException with message "Not all of the required headers are signed"

Reply:
@reta Have incorporated the test case described in the conversation above

Reply:
:+1: thank you, LGTM

Reply:
@reta Path parameters were being decoded while verifying the signatures and hence there were mismatch in the request-target , have provided fix to avoid decoding of the path parameters by using rawpath instead of path

Reply:
@AnilKumarHurkadli thank you, makes sense to me: it is unclear why previously the `uri.getPath()` was combined with `uri.getRawQuery()`, either getPath() + getQuery() or getRawPath() + getRawQuery() should have been followed (imho), thank you for fixing.

Reply:
@reta Thank you for reviewing the changes , yes rawpath and rawquery works well as it does not decodes and retains as is for verification.